src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md

```diff
@@ -12,20 +12,24 @@ James products rely historically on Spring for dependency injection. It doesn't
 James uses Spring in a way that enables overriding any class via a configuration file thus endangering overall correctness by giving too much 
 power to the user.
 
-James propose several implementations for each of the interfaces it defines. The number of possible combinations of implementations is thus really high (like factorial(n) with n > 10). It makes it unpractical to run tests for each possible interface
-combination. We rather run integration tests for combinations that we decide brings the more value to the users. By overriding any class, Spring defeats 
-this testing logic. Knowing which component implementation combines well with which other one is not intuitive to users,
-hard to document and test.
-
-We thus decided to rely on Guice. Components are defined by code in a static fashion. Overriding components requires 
-explicit code modification and recompilation, thus warning the user about the impact of the choices he does, and lowering 
-the project's responsibility.
+James propose several implementations for each of the interfaces we define. The number of possible combinations of
+implementations is thus really high (like factorial(n) with n > 10). It makes it unpractical to run tests for each 
+possible interface combination. We rather run integration tests for combinations that we decide brings the more value to
+the users. We rather run integration tests for combinations that make sense. By overriding any components, and 
+allowing arbitrary component combination, Spring defeats this testing logic.
```

Comment:
It's not clear with this wording why changing classes breaks the principle of "test only combinations that make sense"